src/h5web/visualizations/line/LineVis.tsx

```diff
@@ -27,36 +33,62 @@ function LineVis(props: Props): ReactElement {
     curveType = CurveType.LineOnly,
     showGrid = true,
     scaleType = ScaleType.Linear,
+    abscissas = range(dataArray.size),
   } = props;
 
-  const indexDomain = extendDomain([0, dataArray.size - 1], 0.01);
+  const abscissaToIndex = getValueToIndexScale(abscissas, true);
+
+  const abscissaDomain = useMemo(() => {
+    const rawDomain = getDataDomain(abscissas);
+    return rawDomain ? extendDomain(rawDomain, 0.01) : undefined;
```

Comment:
Since we're checking `abscissaDomain` for `undefined` now, it may make sense for `extendDomain` to support an `undefined` input/output domain: `return extendDomain(getDataDomain(abscissas, 0.01))`

Reply:
A good idea on the paper but is it really in practice ? We will need then to check for `undefined` on every use of `extendDomain`...

Reply:
Ah yeah, forgot it's used in `DomainSlider`... Oh well, you can at least simplify with `rawDomain && extendDomain(...)`